src/common/cluster-store.ts

```diff
@@ -97,8 +107,26 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
     return this.clusters.get(id);
   }
 
-  getByWorkspaceId(workspaceId: string): Cluster[] {
-    return this.clustersList.filter(cluster => cluster.workspace === workspaceId)
+  getByWorkspaceId(workspaceId: string, sorted = true): Cluster[] {
+    const ordering = this.clusterOrders[workspaceId]?.reduce((acc: ClusterIconOrderingMap, cur, i) => {acc[cur] = i; return acc;}, {});
+    const clusters = this.clustersList.filter(cluster => cluster.workspace === workspaceId);
+    if (!ordering || !sorted) {
+      return clusters;
+    }
+
+    const sortedClusters = [];
+    const unsortedClusters = [];
+
+    for (const cluster of clusters) {
+      const index = ordering[cluster.id];
+      if (typeof index === "number" && index >= 0) {
+        sortedClusters[index] = cluster;
+      } else {
+        unsortedClusters.push(cluster);
+      }
+    }
```

Comment:
All this looks too  much complicated for such simple task.
Should be just `return this.clusterOrders[workspace].map(id => this.clusters.get(id))`
And `clusterOrders` will be pre-filled on init, indexes updated on dnd.